contracts/roles/AgentRoles.sol

```diff
@@ -0,0 +1,117 @@
+pragma solidity ^0.6.0;
+
+import "openzeppelin-solidity/contracts/access/Roles.sol";
+import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+
+contract AgentRoles is Ownable {
+    using Roles for Roles.Role;
+
+    event RoleAdded(address indexed account);
+    event RoleRemoved(address indexed account);
+
+    Roles.Role private _supplyModifiers;
+    Roles.Role private _freezers;
+    Roles.Role private _transferManagers;
+    Roles.Role private _recoveryAgents;
+    Roles.Role private _complianceAgents;
+    Roles.Role private _whiteListManagers;
+
+
+    // SupplyModifier Role _supplyModifiers
+
+    function isSupplyModifier(address account) public view returns (bool) {
+        return _supplyModifiers.has(account);
+    }
+
+    function addSupplyModifier(address account) public onlyOwner {
```

Comment:
Maybe we should add a role authorized to update role members instead of leaving it to the Owner only.